src/st/st-theme-node.c

```diff
@@ -610,6 +610,52 @@ get_color_from_rgba_term (CRTerm       *term,
   return VALUE_FOUND;
 }
 
+static GetFromTermResult
+get_color_from_context (CRTerm         *term,
+                        StThemeContext *context,
+                        ClutterColor   *color)
+{
+  CRTerm *arg = term->ext_content.func_param;
+  gchar *name;
```

Comment:
I didn't do a real review or test this. I don't even understand everything you're doing here but this gchar * should probably be freed when you're done with it. Since you only use it in the if() block below I would just declare it there and then free it right after you set the envcolor.

Reply:
That is just a reference to a string in the term struct, which would get freed (I suppose) whether I reference it or not.